### PR TITLE
Feature/support download backup version api

### PR DESCRIPTION
**Pull request by strausr**

### CODE REVIEW

File: lib-es5/api.js [hunk 1]
```diff
@@ -111,20 +111,22 @@ exports.resource = function resource(public_id, callback) {
...
   type = options.type || "upload";
   uri = ["resources", resource_type, type, public_id];
-  return call_api("get", uri, only(options, "exif", "colors", "derived_next_cursor", "faces", "image_metadata", "pages", "phash", "coordinates", "max_results"), callback, options);
+  return call_api("get", uri, only(options, "exif", "colors", "derived_next_cursor", "faces", "image_metadata", "pages", "phash", "coordinates", "max_results", "versions"), callback, options);
 };
 
 exports.restore = function restore(public_ids, callback) {
   var options = arguments.length > 2 && arguments[2] !== undefined ? arguments[2] : {};
 
+  options.content_type = 'json';
```
Comment (asisayag2) [inline lib-es5/api.js:120]: @strausr , why is that needed?

File: test/api_spec.js [hunk 2]
```diff
@@ -953,6 +986,40 @@ describe("api", function () {
       expect(resource).not.to.be(null);
       expect(resource.bytes).to.eql(3381);
     }));
+    it('should restore a deleted resource by versions', function () {
+      this.timeout(TIMEOUT.LARGE);
+      return Q.all([
+        uploadImage({
```
Comment (asisayag2) [inline test/api_spec.js:992]: @strausr , shouldn't we upload 2 versions for the same publicId for checking this scenario?

Comment (strausr) [reply test/api_spec.js:992]: Based on the spec, it's a list that maps public_ids to versions. I don't think that you can map one public_id to many versions. That, being said, I'll check to see if that works. If it does, I'll add a separate test.

File: lib-es5/api.js (hunk 1 above)
Comment (strausr) [reply lib-es5/api.js:120]: This is done to accept an array as part of the options. Meaning this is how we know to parse an array.

File: test/api_spec.js (hunk 2 above)
Comment (asisayag2) [reply test/api_spec.js:992]: I meant that for restoring an **old** version, you need to have at least 2 versions. The old and the new

File: lib-es5/api.js (hunk 1 above)
Comment (asisayag2) [reply lib-es5/api.js:120]: Not sure I get what you mean

Comment (patrick-tolosa) [reply lib-es5/api.js:120]: By default, we send all our requests URL-encoded (even post requests).

It seems our servers are unable (bug, or by design) to parse arrays sent this way.

We don't have this problem once we change the payload to JSON.
We have this line of code littered throughout the project (setting the content_type).

We might want to investigate why we need this (it exists in other SDKs as well)

File: test/api_spec.js (hunk 2 above)
Comment (strausr) [reply test/api_spec.js:992]: added a new test to restore an older version

File: test/api_spec.js [hunk 3]
```diff
@@ -321,6 +325,36 @@ describe("api", function () {
...
+    }));
+    after(function () {
+      return cloudinary.v2.api.delete_resources(publicId).then((response) => {
+        expect(response).to.have.property("deleted");
+      });
+    });
+    it("should return the asset details together with all of its backed up versions when versions is true", function () {
+      return cloudinary.v2.api.resource(publicId, { versions: true })
+        .then((resource) => {
+          expect(resource.versions).not.to.be(undefined);
```
Comment (patrick-tolosa) [inline test/api_spec.js:346]: I guess these are the same no? if 'resource.versions' is an array, it's also not undefined?

Comment (strausr) [reply test/api_spec.js:346]: yeah, I'll remove the undefined one
